### PR TITLE
fix: only show brands with liquidity pools

### LINKED ISSUE #20: Crashes when loading liquidity pools on devnet

### PR DESCRIPTION
**Pull request by samsiegart**
refs: https://github.com/Agoric/dapp-amm/issues/21
fixes: https://github.com/Agoric/dapp-amm/issues/20

Does not fully solve https://github.com/Agoric/dapp-amm/issues/21 because it will still crash when getting the price for an asset if the pool exists but has 0 liquidity in it. This should be handled separately when we use price notifiers though.

Fixes https://github.com/Agoric/dapp-amm/issues/20 by hiding liquidity pools with 0 liquidity. Their associated `Central` and `Secondary` properties are null which was causing an error trying to render them.

### CODE REVIEW

File: ui/src/components/Liquidity/LiquidityPool/BodyLiquidityPool.jsx [hunk 1]
```diff
@@ -32,26 +32,28 @@ const BodyLiquidityPool = props => {
...
-        PLACES_TO_SHOW,
-      );
-      return {
-        Central: { info: centralInfo, value: centralValString },
-        Secondary: { info: secondaryInfo, value: secondaryValString },
-      };
-    });
+    return pool?.allocations
+      ?.filter(({ Central, Secondary }) => Central && Secondary)
+      ?.map(item => {
```
Comment (turadg) [inline ui/src/components/Liquidity/LiquidityPool/BodyLiquidityPool.jsx]: `filter` returns an array.
```suggestion
      .map(item => {
```

File: ui/src/components/Swap/DialogSwap/AssetDialog.jsx [hunk 2]
```diff
@@ -1,87 +1,26 @@
...
-    refinedLiquidityBrands(assets);
-
-    return () => {
-      mounted.current = false;
-    };
-  }, []);
+  const filteredAssets =
+    assets?.filter(
+      ({ brand }) => poolBrands.includes(brand) || brand === centralBrand,
+    ) ?? [];
```
Comment (turadg) [inline ui/src/components/Swap/DialogSwap/AssetDialog.jsx]:
```suggestion
  const filteredAssets = (assets || []).filter(
      ({ brand }) => poolBrands.includes(brand) || brand === centralBrand,
    );
```

File: ui/src/components/Swap/DialogSwap/AssetDialog.jsx [hunk 3]
```diff
@@ -96,12 +35,12 @@ const AssetDialog = ({ type, setSelectedAsset }) => {
 
   return (
     <div className="flex flex-col gap-4 p-5 overflow-auto ">
-      {parsedAssets?.map(item => (
+      {filteredAssets?.map(item => (
```
Comment (turadg) [inline ui/src/components/Swap/DialogSwap/AssetDialog.jsx]: it's defined as an array
```suggestion
      {filteredAssets.map(item => (
```

**Review by turadg — APPROVED** [review]

File: ui/src/components/Liquidity/LiquidityPool/BodyLiquidityPool.jsx (hunk 1 above)
Comment (samsiegart) [reply ui/src/components/Liquidity/LiquidityPool/BodyLiquidityPool.jsx]: Done

File: ui/src/components/Swap/DialogSwap/AssetDialog.jsx (hunk 2 above)
Comment (samsiegart) [reply ui/src/components/Swap/DialogSwap/AssetDialog.jsx]: Done

File: ui/src/components/Swap/DialogSwap/AssetDialog.jsx (hunk 3 above)
Comment (samsiegart) [reply ui/src/components/Swap/DialogSwap/AssetDialog.jsx]: Done

**Pull request merged**